### DISCUSSION TITLE
Atoms with opposite chirality are matched in rdMCS.FindMCS

### DISCUSSION THREAD

**Discussion by laurenreid1** [rdkit/rdkit#4752] [Q&A]
Hi,

I'm trying to compute the MCSS of molecules that may have chiral centres of either stereochemistry. In the cases where the stereochemistry is opposite, I would like to exclude the chiral atoms from the MCSS. For example, the following would ideally return an MCSS of benzene, since the chiral centres don't match:

```
mol1 = Chem.MolFromSmiles("[C@@](c1ccccc1)(C1CCC1)(N)(c1[nH]ccc1)")
mol2 = Chem.MolFromSmiles("[C@](c1ccccc1)(C1CCC1)(N)(c1[nH]ccc1)")

mcss = rdFMCS.FindMCS([mol1, mol2], matchChiralTag=True)

print(Chem.MolToSmiles(mcss.queryMol))
```

However, the output is as follows:

`*1~*~*~*(~*(~*2~*~*~*~*~2)~*2~*~*~*~2)~*~*~1`

I can see that removing the smallest chiral attachment (the N) makes the central carbon achiral while retaining the maximum number of atoms, but is this the desired output of MCSS since the chiral atoms in the original molecules shouldn't be matched?

I think this is related to issue: https://github.com/rdkit/rdkit/issues/2624#issue-482352937

Does anyone know how I can alter this for my desired functionality?

Thanks in advance for the help,

Lauren

Python version: 3.7.7
RDKit version: 2020.03.3

**Comment by greglandrum** [comment]
Hi @laurenreid1,
Sorry for the slow response.
The way `matchChiralTag` works is definitely somewhat confusing, but I will try and explain my understanding of what it does.

The default behavior of `FindMCS()`, which is `matchChiralTag=False`, is to completely ignore stereochemistry.
If `matchChiralTag` is True, there are two changes to the algorithm:
1. Atoms which have chirality specified can only match other atoms which have chirality specified
2. If the MCS includes at least three atoms around an atom with chirality specified, then the chirality must match in all molecules.

To demonstrate this, here's an example with three molecules that differ only in the stereochemistry at a single atom:
```
In [27]: ms = [Chem.MolFromSmiles(x) for x in ('C[C@H](F)Cl','C[C@@H](F)Cl','CC(F)Cl')]
```
So that's one molecule with `@`, one with `@@` and one where the stereo is unspecified.

The default MCS is what you'd expect - every atom matches:
```
In [29]: rdFMCS.FindMCS(ms).smartsString
Out[29]: '[#6]-[#6](-[#9])-[#17]'
```

If we do the MCS of the first molecule and the last one using `matchChiralTag`, we only get a single atom (which atom you get is not _a priori_ obvious):
```
In [32]: rdFMCS.FindMCS((ms[0],ms[2]),matchChiralTag=True).smartsString
Out[32]: '[#17]'
```
This is because the central C atom doesn't match anymore since it has specified chirality in the first molecule and unspecified chirality in the last.

If, on the other hand, we do the MCS of the 1st and 2nd molecules, we get a three atom result:
```
In [31]: rdFMCS.FindMCS((ms[0],ms[1]),matchChiralTag=True).smartsString
Out[31]: '[#6]-[#6]-[#17]'
```
Here the central carbons can match since they have chirality specified in both molecules. But we can only match up to two neighbors of the central carbon since the moment we add a third neighbor the actual chirality itself starts to matter.

Note that there are multiple possible three-atom answers and it's not _a priori_ obvious which you will get.

Finally, going back to the full set, if we turn on matchChiralTag the MCS is a single atom (which atom you get isn't obvious):
```
In [30]: rdFMCS.FindMCS(ms,matchChiralTag=True).smartsString
Out[30]: '[#6]'
```

Does that help explain it?

**Reply by laurenreid1** [reply]
Hi Greg,

Thank you for the detailed explanation.

That makes sense, I can understand that retaining a maximum of 2 chiral attachments would restrict the mcss to be planar. However, I find this behaviour changes when the chiral H is substituted for an explicit non-hydrogen group. So altering your example to:

```
ms = [Chem.MolFromSmiles(x) for x in ('C[C@](N)(F)Cl','C[C@@](N)(F)Cl','CC(N)(F)Cl')]
rdFMCS.FindMCS((ms[0],ms[1]),matchChiralTag=True).smartsString
```

gives:

`'[#6](-[#7])(-[#9])-[#17]'`

which includes 3 of the chiral carbon neighbours. Is there a reason this alternate behaviour should occur with a non-hydrogen group?

Thanks,

Lauren

**Comment by greglandrum** [comment]
Lauren,
That's an excellent question, but I don't have a quick answer for it. I will see if I can track that down. Unless @ptosco has an idea?

**Comment by ptosco** [comment]
Unfortunately I don't, I need to dig a bit to figure out. Happy to, but only after 17 Dec :-)

**Reply by laurenreid1** [reply]
Thank you for taking a look :)

Lauren
